setup.cfg

```diff
@@ -120,14 +120,14 @@ test =
 dev =
     # test dependencies are required for developing localstack
     %(test)s
-    black==23.10.0
+    black>=23.10.0
     coveralls>=3.3.1
     Cython
     networkx>=2.8.4
     pandoc
-    pre-commit==3.5.0
+    pre-commit>=3.5.0
     pypandoc
-    ruff==0.1.0
+    ruff>=0.1.0
```

Comment:
Looks like some of the main issues are with the linting being different now. I'd say we keep those pins until we decide on doing the switch to `ruff format`. Then we can remove `black`, upgrade `ruff` and do the formatting in one step. 

There will still be the question however what happens with the formats in the future. If there is some big change in `ruff` which would affect a lot of files, the dependency pin upgrade PR would contain a lot of these in order to be green. We would then potentially need to add a lot of those upgrade commits to the `.git-blame-ignore-revs` list.

Reply:
Generally speaking yes, if an upgrade needs a proper "fix", it should be done outside of the update PR (where we would avoid this version in the `setup.cfg` in the update PR, and remove the pin in the "fix PR"). If it's just a few lines, an "adjustment" to the new version, I would do it in the upgrade PR itself.
We actually already do it like this with the ASF updates. Small adjustments (adjusting the provider method signatures to the changed ASF API stubs) are done in the ASF update PRs directly, while bigger fixes related to upgrades (like the sqs protocol handling) are done separately.

Reply:
So to clarify with the `black` update:
The further procedure if we continue with `black` and want to update it would be:
- Pin the `black` version to the latest one that worked and merge it like that.
- Create a PR where only the `black` pin is removed again
- In that PR regenerate the pins with the newest `black` version
- Fix the issues in the new PR (in this case just the reformat)

Correct?

Reply:
Another thing that comes to mind here is: How can we ensure that pins that get introduced because of issues like that will get resolved in a timely manner? Only setting a TODO might get lost over time, like some of the pins we already had in there. Maybe we should have it as a rule that immediately after merging an upgrade PR with a pin, we should create a PR that removes the pin again and keep it around until the proper fix happens either upstream or by us doing the fix in the PR.